### ISSUE TITLE
[BUG] ConvTimeNetForecaster silently ignores random_state=0 due to falsy check

### ISSUE THREAD

**Issue by paramsureliya** [alan-turing-institute/sktime#10052]
### Describe the bug

`ConvTimeNetForecaster.build_pytorch_train_dataloader` and its pred
dataloader use `if self.random_state:` to check whether to seed the
PyTorch Generator. This is a falsy check `random_state=0` evaluates
to `False`, so seed 0 is silently ignored and the generator runs without
a fixed seed.

This means results are not reproducible even though the user explicitly
passed `random_state=0`, with no error or warning raised.

Seed 0 is extremely common in research and tutorials.

### Location

`sktime/forecasting/convtimenet.py` lines 355 and 409

### Buggy code

```python
# line 355 and 409 - both dataloaders
if self.random_state:              # 0 is falsy - seed never set
    gen.manual_seed(self.random_state)
```

### Fix

```python
if self.random_state is not None:  # correctly handles seed 0
    gen.manual_seed(self.random_state)
```

### Note

The network class `networks/convtimenet/forecaster/_convtimenet.py`
and the classification base `classification/deep_learning/_pytorch.py`
already use the correct `is not None` pattern. This fix makes the
forecaster consistent with the rest of the codebase.

### Impact

Any user passing `random_state=0` gets non-reproducible results silently.
